pkg/model/config/config.go

```diff
@@ -78,39 +81,105 @@ func (c Config) IsV3() bool {
 	return c.Version == Version3Alpha
 }
 
-// HasResource returns true if API resource is already tracked
-func (c Config) HasResource(target GVK) bool {
-	// Return true if the target resource is found in the tracked resources
+type unsupportedFieldError struct {
+	fieldName string
+	version   string
+}
+
+func (e unsupportedFieldError) Error() string {
+	return fmt.Sprintf("`%s` field found for %s", e.fieldName, e.version)
+}
+
+func (c Config) errorForField(fieldName string) error {
+	return unsupportedFieldError{fieldName: fieldName, version: c.Version}
+}
+
+// validate checks if a Config is valid or it has some fields that shouldn't be present for that version
+func (c Config) validate() error {
+	if c.IsV2() {
+		if c.ProjectName != "" {
+			return c.errorForField("projectName")
+		}
+		for _, r := range c.Resources {
+			if r.Domain != "" {
+				return c.errorForField("resources[].domain")
+			}
+			if r.Plural != "" {
+				return c.errorForField("resources[].plural")
+			}
+			if r.Path != "" {
+				return c.errorForField("resources[].plural")
```

Comment:
```suggestion
				return c.errorForField("resources[].path")
```

Reply:
endpoint.

Reply:
Path makes more sense. Why do you suggest endpoint?

Reply:
We are here renaming package to path because `k8s.io/api/apps/v1` is not really a go pkg. It is the Rest Endpoint for the API. So, IMO path is not a good name either. Because what we are storing here are API endpoints and it shows more aligned indeed with k8s docs e.g [Learn about API endpoints, resource types and samples by reading API Reference](https://kubernetes.io/docs/concepts/overview/kubernetes-api/)

Yesterday, spoking with @Adirio we agreed on to move forward with the endpoint.

Reply:
Agree with you in another comment that Path is really a better name indeed.